KIPs/kip-xx.md

```diff
@@ -0,0 +1,57 @@
+---
+kip: to be assigned
+title: Transaction data gas cost reduction
+author: Lake (@hyunsooda) and Ollie (@blukat29)
+discussions-to: https://github.com/kaiachain/kips/issues/22
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-10-18
+---
+
+## Abstract
+
+Reduce the calldata gas costs to the [EIP-2028](https://eips.ethereum.org/EIPS/eip-2028) level. The purpose is to achieve better compatibility with Ethereum-oriented SDKs and tools.
+
+## Motivation
+
+Since the introduction of [EIP-2028](https://eips.ethereum.org/EIPS/eip-2028), several bundler software solutions have been developed.
```

Comment:
EIP-4337 is more relevant I suppose? https://eips.ethereum.org/EIPS/eip-4337#-eth_estimateuseroperationgas

EIP-2028 just happens to be the latest gas formula when the 4337 was born. So 2028 is not directly related to bundlers.

I think the logic goes like

- Since the introduction of EIP-4337, bundlers developed
- One of its feature, `eth_estimateUserOperationGas` is adjusted to the latest EIP-2028 gas schedule. Note that EIP-2028 has been used in Ethereum for a long time since its Istanbul hardfork.
- So bundlers are incompatible with Kaia's gas schedule without explicit configuration. It hinders smooth integration of 4337 bundlers.
- With this proposal, Kaia mainnet will align with EIP-2028 since Kaia's Prague hardfork.

Reply:
> EIP-4337 is more relevant I suppose?

Right. Fixed.

> I also don't want to name the hardfork just yet. Let's just say "the new hardfork" in this proposal.

Changed.

And the listed comments seem to be fit already. If not, please let me the detail again. Thanks

Reply:
This is how I would say it.

```
The adoption of EIP-2028 will improve the compatibility with ERC-4337 ecosystem tools.

Since the introduction of [ERC-4337](https://eips.ethereum.org/EIPS/eip-4337), several bundler software solutions have been developed. The bundler's `eth_estimateUserOperationGas` API calculates the gas required for a UserOperation. The calculation depends on the calldata costs in a UserOperation. The issue is that some tools assumes EIP-2028 rule [unless explicitly configured](https://github.com/eth-infinitism/bundler/blob/f4647969386aa859b7edf608467168d140e5f92c/packages/sdk/src/PreVerificationGasCalculator.ts#L44-L54), making their result incompatible with Kaia.

This proposal introduces a new hardfork that aligns the calldata cost calculation to that of EIP-2028.
```

To me nonzero vs. zero is not the most important information. I'd rather accurately and concisely refer to the EIP-2028.

Reply:
Got it. I understand that ecosystem alignment should be highlighted rather than a new cost tag. https://github.com/kaiachain/kips/pull/23/commits/9fd34bf9cf1570d5ec055bc7474c68b4c9e4e891